service/models.py

```diff
@@ -67,22 +66,33 @@ def deserialize(self, data):
 
     @classmethod
     def init_db(cls, app):
-        """ Initializes the database session """       
+        """ Initializes the database session """
         cls.logger.info('Initializing database')
         cls.app = app
         # This is where we initialize mongoDB from the Flask app
         # db.init_app(app)
         db = connect('myDatabase')
         app.app_context().push()
         # db.create_all()
-    
+
     @classmethod
     def all(cls):
         #This is a function to return all suppliers
         cls.logger.info('Processing all suppliers')
         return cls.objects()
 
 
+    @classmethod
+    def find(cls, supplier_id):
+        """Retrieves a single supplier with a given id (supplierID) """
+
+        cls.logger.info('Getting supplier with id: %s'.format(supplier_id))
+
+        try:
+            return cls.objects(id=supplier_id)
```

Comment:
**cls.objects(id=supplier_id)** returns a list of suppliers containing only one object which holds the required **supplier_id**. Therefore maybe better just return the only object rather than an array? 😏

Can be done as **cls.objects(id=supplier_id).first()**